### PR TITLE
Import remaining docs from wiki

### PR DESCRIPTION
**Pull request by raghavendra-talur**
# What does this PR achieve? Why do we need it?
There were a few more docs remaining in the wiki which were not imported under the docs dir. I have now imported them as-is.

### Notes for the reviewer
There are more things to be edited because some the documentation in the wiki had gone stale. I will send a new PR with changes after this PR is merged as I want to restrict this PR to import function.

### CODE REVIEW

File: docs/design/architecture.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Overview
```
Comment (phlogistonjohn) [inline docs/design/architecture.md]: I think we should either drop this doc or defer it to another PR that updates the content as it is woefully out of date. It isn't entirely wrong but it's missing a lot of relevant detail and emphasizes parts of the architecture that are minor while not touching on important aspects.

**Comment by phlogistonjohn** [comment]
Any progress on this?

File: docs/design/architecture.md (hunk 1 above)
Comment (raghavendra-talur) [reply docs/design/architecture.md]: Removed the commit that added architecture.md file.

**Comment by raghavendra-talur** [comment]
@centos-ci retest this please

**Comment by phlogistonjohn** [comment]
@raghavendra-talur as this is a doc (and not code change) do you wish to try and finish this effort or shall we close this?

**Comment by iamniting** [comment]
I will leave this up to @raghavendra-talur if he thinks that efforts are very minimal and can be done easily then let's get it in otherwise let's close it.

**Comment by raghavendra-talur** [comment]
@phlogistonjohn I have removed the extra changes that were done in the PR. The changes we have now are
* we only import testing and release related pages from the wiki
* we remove the deadlinks
*

**Review by phlogistonjohn — APPROVED** [review]
I was a tad confused at first by the removal of workflow subsection but you had a great commit message the explained it before I needed to ask you. :-)
Everything else looks fine.

**Pull request merged**